### ISSUE TITLE
Search and Restore deleted items

### ISSUE THREAD

**Issue by psychowood** [firefly-iii/firefly-iii#3728]
I noticed that firefly-iii does not physically delete items, but rather has a deleted_at column on most tables, and that is very nice :)

Do you have any plans to use that field? Do the deleted items get periodically cleaned up/trashed?

I was thinking of a search filter (":deleted" ?)  to search also for deleted items, replacing the action with just a restore/undo action.

<img width="173" alt="image" src="https://user-images.githubusercontent.com/115389/91037382-36667580-e609-11ea-8eb2-cfb2aa5231f5.png">

It's probably an overkill to allow viewing deleted transactions before restoring them (API, UI, backend implications), maybe an alternative could be allowing cloning instead of restoring, from actions menu in the search list?

**Comment by JC5** [comment]
No that's not going to work. Most information gets saved when you press delete, but not all of it. Whatever you restore it's going to be a partial save and that's just more work. Transaction/budget/category/tag links are gone, some meta data may be deleted, not everything is saved.

The major objects like accounts, transactions etc have a "deleted_at" field so database integrity can be verified more easily (and corrected if necessary). Imagine your DB fails half-way a deletion. That can be corrected later on because the soft delete allows you to backtrack what happened. There is no code to do this btw, it never happens.

But links and connections and intermediate table information is never saved. This makes a restore less than optimal.

In short: if you delete something, despite all the big fat warnings, you'll be glad to see *most* information is still in the DB. But the fastest way to a restoration is either recreating the info (using the deleted item as a check) or restoring your DB entirely.

**Comment by psychowood** [comment]
Understood, thanks!
